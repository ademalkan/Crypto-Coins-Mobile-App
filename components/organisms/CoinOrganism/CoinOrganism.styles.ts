import { Dimensions, StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    padding: 10,
    width: '100%',
    borderBottomWidth: 1,
    borderBottomColor: '#f2f2f2',
  },
  image: {
    width: 70,
    height: 70,
    marginRight: 10,
    // backgroundColor: "red",
  },
  options: {
    display: "flex",
    flex: 1,
    alignItems: "center",
    flexDirection: "row",
    justifyContent: "space-between",
  }



})