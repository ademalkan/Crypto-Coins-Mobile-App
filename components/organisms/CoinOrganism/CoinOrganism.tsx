import React from 'react';
import { Image } from 'expo-image';

import { Text, TouchableOpacity, View } from 'react-native';
import styles from "./CoinOrganism.styles";
import Price from '../../atoms/Price/Price';
import Symbol from '../../atoms/Symbol/Symbol';
import Name from '../../atoms/Name/Name';
import { useNavigation } from 'expo-router';
import { Button } from '../../atoms/Button/Button';

type CryptoCoinProps = {
  name: string;
  symbol: string;
  price: number;
  image: string;
  id: string;
};

const CoinOrganism: React.FC<CryptoCoinProps> = ({ name, symbol, price, image, id }) => {
  const navigation = useNavigation();

  const handlePress = () => {
    navigation.navigate('CoinDetail', { coin: id });
  };

  return (
    <View style={styles.container}>
      <TouchableOpacity onPress={handlePress}>
        <Image
          style={styles.image}
          source={require('./../../../assets/images/logo.png')}
          contentFit="cover"
          transition={1000}
        />
      </TouchableOpacity>

      <View>
        <Name name={name} />
        <Symbol symbol={symbol} />
        <View style={styles.options}>
          <Button onPress={handlePress} text="Detail" />
          <Button style={{ marginLeft: 10 }} onPress={handlePress} text="Add Favorite" />
        </View>
      </View>

      <Price price={price} />
    </View>
  );
};

export default CoinOrganism;
