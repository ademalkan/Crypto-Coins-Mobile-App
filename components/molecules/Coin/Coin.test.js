import React from "react";
import renderer from "react-test-renderer";
import CoinOrganism from "./Coin";

jest.mock("expo-router", () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

describe("CoinOrganism", () => {
  it("should render the coin organism correctly", () => {
    const coin = {
      name: "Bitcoin",
      symbol: "BTC",
      price: 50000,
      image: "coin.png",
      id: "123",
      total_volume: 1000000,
      price_change_24h: 5,
    };

    const component = renderer.create(<CoinOrganism {...coin} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
