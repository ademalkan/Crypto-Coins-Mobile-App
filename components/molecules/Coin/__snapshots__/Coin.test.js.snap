// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinOrganism should render the coin organism correctly 1`] = `
<View
  style={
    {
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <ViewManagerAdapter_ExpoImage
      height={70}
      marginRight={10}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      proxiedProperties={
        {
          "contentFit": "cover",
          "contentPosition": {
            "left": "50%",
            "top": "50%",
          },
          "placeholder": [],
          "source": [],
          "tintColor": undefined,
          "transition": {
            "duration": 1000,
          },
        }
      }
      style={
        {
          "height": 70,
          "marginRight": 10,
          "width": 70,
        }
      }
      width={70}
    />
  </View>
  <View>
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
          "marginRight": 5,
        }
      }
    >
      Bitcoin
    </Text>
    <Text
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "textTransform": "uppercase",
        }
      }
    >
      BTC
    </Text>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginBottom": 4,
            "textTransform": "capitalize",
          }
        }
      >
        Total Volume:
      </Text>
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 14,
            "fontWeight": "500",
            "marginLeft": 4,
          }
        }
      >
        $
        1000000
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginBottom": 4,
            "textTransform": "capitalize",
          }
        }
      >
        Price Change 24 Hours:
      </Text>
      <Text
        style={
          {
            "color": "#666",
            "fontSize": 14,
            "fontWeight": "500",
            "marginLeft": 4,
          }
        }
      >
        $
        5
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
          "width": 205,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#383a68",
            "borderRadius": 8,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
              },
              undefined,
            ]
          }
        >
          Detail
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#383a68",
            "borderRadius": 8,
            "opacity": 1,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
              },
              undefined,
            ]
          }
        >
          Add Favorite
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
        "marginLeft": "auto",
      }
    }
  >
    $
    50000
  </Text>
</View>
`;
